lib/mauth/client.rb

```diff
@@ -265,19 +282,40 @@ def signed(object, attributes = {})
       end
 
       # takes a signable object (outgoing request or response). returns a hash of headers to be
-      # applied tothe object which comprise its signature.
+      # applied to the object which comprise its signature.
       def signed_headers(object, attributes = {})
+        v1_only_override = attributes.delete(:v1_only_override)
+        v2_only_override = attributes.delete(:v2_only_override)
+
+        if v1_only_override # used when signing responses to requests with only the v1 protocol, when AUTHENTICATE_WITH_ONLY_V2 is false and SIGN_REQUESTS_WITH_ONLY_V2 is true
+          signed_headers_v1(object, attributes)
+        elsif sign_requests_with_only_v2? || v2_only_override # override used when signing responses to requests with the v2 protocol when SIGN_REQUESTS_WITH_ONLY_V2 is false
+          signed_headers_v2(object, attributes)
+        else # by default sign with both the v1 and v2 protocol
```

Comment:
so when `v2_only_override: true` is set then that means we do both v1 and v2, does not sounds as much of an override

Reply:
when `v2_only_override: true` we only sign with v2.